internal/cli/opsmanager/serverusage/download.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverusage
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+type DownloadOpts struct {
+	cli.DownloaderOpts
+	startDate string
+	endDate   string
+	format    string
+	store     store.ServerUsageReportDownloader
+}
+
+var downloadMessage = "Download of %s completed.\n"
+
+func (opts *DownloadOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *DownloadOpts) Run() error {
+	f, err := opts.NewWriteCloser()
+	if err != nil {
+		return err
+	}
+
+	if err := opts.store.DownloadServerUsageReport(opts.newServerTypeOptions(), f); err != nil {
+		_ = opts.OnError(f)
+		return err
+	}
+	if opts.Out != "/dev/stdout" {
+		fmt.Printf(downloadMessage, opts.Out)
+	}
+	return f.Close()
+}
+
+func (opts *DownloadOpts) newServerTypeOptions() *opsmngr.ServerTypeOptions {
+	return &opsmngr.ServerTypeOptions{
+		StartDate:  opts.startDate,
+		EndDate:    opts.endDate,
+		FileFormat: opts.format,
+	}
+}
+
+func (opts *DownloadOpts) initDefaultOut() {
+	if opts.Out == "" {
+		opts.Out = "serverUsageReport" + opts.format
```

Comment:
having resolved the format, could we improve the naming? the UI returns a name with the date filters in it, "serverUsage-{start-date}-{end-date}.{format}" can we do something similar here please? it makes it easier for users avoid to have to set names if creating different files for different date ranges, also with my other comment, we should not need to pass the dot(.) on the format, that's a naming problem you can solve here

Reply:
sure 👍